server/src/main/java/com/cloud/api/query/QueryManagerImpl.java

```diff
@@ -971,6 +972,17 @@ public ListResponse<UserVmResponse> searchForUserVMs(ListVMsCmd cmd) {
         return response;
     }
 
+    private Object getObjectPossibleMethodValue(Object obj, String methodName) {
+        Object result = null;
+
+        try {
+            Method m = obj.getClass().getMethod(methodName);
+            result = m.invoke(obj);
+        } catch (NoSuchMethodException | InvocationTargetException | IllegalAccessException ignored) {}
```

Comment:
Perhaps log here? @DaanHoogland or do you think we may never hit this?

Reply:
i was thinking that as well, but didn´t want to change to much anymore. We will hit this for sure in the current state if ListVMsCmd is called in the API. It will try to add the attributes that don´t exist and we are supposed to ignore those.
```suggestion
        } catch (NoSuchMethodException | InvocationTargetException | IllegalAccessException ignored) {/*ignored when non-ByAdmin are calling this.*/}
```